src/modules/water-demand-prediction/water-demand-prediction.component.ts

```diff
@@ -0,0 +1,658 @@
+  signal,
+  ViewChildren,
+  Component,
+  OnInit,
+  QueryList,
+} from "@angular/core";
+import {TranslatePipe} from "@ngx-translate/core";
+import {ChartConfiguration, ChartData, ChartDataset, ChartType} from "chart.js";
+import {BaseChartDirective} from "ng2-charts";
+
+import {
+  PredictedSmartmeterDataset,
+  SmartmeterDataset,
+  PredictionSingleSmartmeter,
+  SingleSmartmeter,
+} from "./water-demand-prediction.interface";
+import {WaterDemandPredictionService} from "../../api/water-demand-prediction.service";
+import {DropdownComponent} from "../../common/components/dropdown/dropdown.component";
+
+@Component({
+  selector: "water-demand-prediction",
+  imports: [BaseChartDirective, DropdownComponent, TranslatePipe, CommonModule],
+  templateUrl: "./water-demand-prediction.component.html",
+  styles: ``,
+})
+export class WaterDemandPredictionComponent implements OnInit {
+  /** the displayed resolution in the charts of real data */
+  displayedResolution = signal<string>("hourly");
+
+  /** variables startpoint dropdown */
+  menuStartPoint = "water-demand-prediction.startpoint.menu";
+  optionsStartPoint: Record<string, string> = {
+    "2021-05-26 00:00:00": "water-demand-prediction.startpoint.options.a",
+    "2021-06-01 00:00:00": "water-demand-prediction.startpoint.options.b",
+    "2022-01-01 00:00:00": "water-demand-prediction.startpoint.options.c",
+  };
+  choiceStartPoint = signal<string>("");
+
+  /** variables resolution dropdown */
+  menuResolution = "water-demand-prediction.choice.resolution";
+  optionsResolution: Record<string, string> = {
+    hourly: "water-demand-prediction.resolution.hourly",
+    daily: "water-demand-prediction.resolution.daily",
+    weekly: "water-demand-prediction.resolution.weekly",
+  };
+  choiceResolution = signal<string>("");
+
+  /** variables timeframe dropdown */
+  menuTime = "water-demand-prediction.choice.timeframe";
+  optionsTime: Record<string, string> = {
+    "one day": "water-demand-prediction.timeframe.one-day",
+    "one week": "water-demand-prediction.timeframe.one-week",
+    "one month": "water-demand-prediction.timeframe.one-month",
+    "three months": "water-demand-prediction.timeframe.three-months",
+    "six months": "water-demand-prediction.timeframe.six-months",
+    "one year": "water-demand-prediction.timeframe.one-year",
+    all: "water-demand-prediction.timeframe.all",
+  };
+  choiceTime = signal<string>("");
+
+  /** variables name dropdown */
+  menuSmartmeter = "water-demand-prediction.choice.smartmeter";
+  optionsSmartmeter: Record<string, string> = {};
+  choiceSmartmeter = signal<string>("");
+
+  menuWeather = "water-demand-prediction.choice.weather";
+  optionsWeather: Record<string, string> = {
+    plain: "water-demand-prediction.weather.plain",
+    air_temperature: "water-demand-prediction.weather.air_temperature",
+    precipitation: "water-demand-prediction.weather.precipitation",
+    moisture: "water-demand-prediction.weather.moisture",
```

Comment:
Why are these part of this typescript file, usually translations strings are handled in the html.